src/content/config.ts

```diff
@@ -1,6 +1,15 @@
-import { defineCollection } from 'astro:content';
-import { docsSchema } from '@astrojs/starlight/schema';
+import { defineCollection, z } from 'astro:content'
+import { docsSchema } from '@astrojs/starlight/schema'
+import { validRange } from 'semver'
 
 export const collections = {
-	docs: defineCollection({ schema: docsSchema() }),
-};
+	docs: defineCollection({
+		schema: docsSchema({
+			extend: z.object({
+				astroRange: z
+					.custom<string>(range => typeof range === 'string' && validRange(range))
```

Comment:
For better error messages:

```suggestion
					.string()
					.refine(validRange, {message: 'Must be a valid semver range'})
```